### ISSUE TITLE
Hardcoded values instead of variables

### ISSUE THREAD

**Issue by lyubomir-popov** [canonical-web-and-design/maas-ui#121]
The new maas-ui repo is relying consistently on hardcoded values, with no references to any of the vanilla variables. In some places there are at least //[TODO](https://github.com/canonical-web-and-design/maas-ui/blob/bafc707487ccfadabe0b9fb9e496535f99dfbe87/ui/src/scss/_patterns_navigation.scss#L61) notes to replace with variables.

In others it is just [magic](https://github.com/canonical-web-and-design/maas-ui/blob/bafc707487ccfadabe0b9fb9e496535f99dfbe87/ui/src/scss/_patterns_table-actions.scss#L3) [numbers](https://github.com/canonical-web-and-design/maas-ui/blob/bafc707487ccfadabe0b9fb9e496535f99dfbe87/ui/src/scss/_patterns_table.scss#L3).

If these fix vanilla problems, please create an issue on the vanilla repo so I can later address and remove the override. 

If they are newly built components, please use the [vanilla spacing variables](https://github.com/canonical-web-and-design/vanilla-framework/blob/d313e92a35cdddb310f66c5a3efa5d2a194a0585/scss/_settings_spacing.scss#L73), so we can make things consistent.

**Comment by lyubomir-popov** [comment]
@Caleb-Ellis nav looks good, just a couple of comments:
missing .p-navigation__row (it is required now https://docs.vanillaframework.io/patterns/navigation/), which causes the nav to be full-width instead of fixed width:
![image](https://user-images.githubusercontent.com/2741678/70152294-d4de2f80-16a4-11ea-9ed8-aba73d75dd01.png)

The lines in the mobile view are from the white theme. Since the nav is dark, you can set the nav to dark by default using `$theme-default-nav: 'dark' !default;`, and delete any dark styling you might have from older incarnations of the nav.

![image](https://user-images.githubusercontent.com/2741678/70152264-c8f26d80-16a4-11ea-865a-aaecdfcb1898.png)

**Comment by lyubomir-popov** [comment]
Apart from the nav, there are still some hardcoded rem values in other places: 
![image](https://user-images.githubusercontent.com/2741678/70152974-0e636a80-16a6-11ea-8700-2af69d220f93.png)
![image](https://user-images.githubusercontent.com/2741678/70152990-158a7880-16a6-11ea-947b-df25af9fb7d3.png)

Some of them seem to be doing things that are available in vanilla, e.g. this replicates the shallow strip:
![image](https://user-images.githubusercontent.com/2741678/70153073-3d79dc00-16a6-11ea-9101-771806b9a2b5.png)

And finally just a heads up, this is fixed in the upcoming v2.5.0 release:
// 20-08-2019 Caleb: Dense tables aren't spaced properly
// https://github.com/canonical-web-and-design/vanilla-framework/issues/2504
```
td:not(.p-table-expanding__panel) [class*="p-button"] {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}
```

So maybe park this and revisit after the release?

**Comment by Caleb-Ellis** [comment]
@lyubomir-popov ah wow my bad, I actually only looked in the scss folder, not in each component folder. We've got more custom stuff than I realised. Give me a sec to fix them up.
